src/lib/coaler/embedder/ConformerEmbedder.cpp

```diff
@@ -89,4 +137,118 @@ namespace coaler::embedder {
         params.clearConfs = false;
         return params;
     }
+
+    /*----------------------------------------------------------------------------------------------------------------*/
+
+    std::vector<multialign::PoseID> ConformerEmbedder::generateNewPosesForAssemblyLigand(
+        RDKit::ROMol *worstLigandMol, const multialign::LigandVector &targets,
+        const std::unordered_map<multialign::LigandID, multialign::PoseID> &conformerIDs) {
+        std::vector<unsigned> newIds;
+        for (const multialign::Ligand &target : targets) {
+            // find mcs
+            const multialign::LigandID targetID = target.getID();
+            const multialign::PoseID targetConformerID = conformerIDs.at(targetID);
+            const RDKit::ROMol targetMol = target.getMolecule();
+            const RDKit::Conformer &targetConformer = targetMol.getConformer(targetConformerID);
+
+            std::vector<RDKit::ROMOL_SPTR> mols = {
+                boost::make_shared<RDKit::ROMol>(*worstLigandMol),
+                boost::make_shared<RDKit::ROMol>(target.getMolecule()),
+            };
+
+            auto mcsResult = get_mcs(mols, false);
+            if (mcsResult.QueryMol == nullptr) {
+                spdlog::warn("no mcs found");
```

Comment:
this should be a none case right? we'll fail on that in the first core calculation. So I guess we should throw here instead of  continue